ajax.php

```diff
@@ -44,23 +44,25 @@ function ajaxAbort($status = 400, $msg = null)
 }
 // --- support functions ------------------
 
-
-
 if (!isset($_GET['act']) || !isset($_GET['method'])) {
     ajaxAbort();
 }
 
 $language_page_directory = DIR_WS_LANGUAGES . $_SESSION['language'] . '/';
 
 $className = 'zc' . ucfirst($_GET['act']);
-$classFile = $className . '.php';
-$basePath  = DIR_FS_CATALOG . DIR_WS_CLASSES;
-
-if (!file_exists(realpath($basePath . 'ajax/' . basename($classFile)))) {
-    ajaxAbort();
+$classFile = basename($className . '.php');
+$classPath = DIR_WS_CLASSES . 'ajax/';
+$basePath  = DIR_FS_CATALOG;
+if (file_exists(realpath($basePath . $classPath . $classFile))) {
```

Comment:
file_exists demands a string, and realpath may return false.

Reply:
Ref #6500